capstone/src/main/java/com/google/sps/servlets/profile/ProfileServlet.java

```diff
@@ -86,32 +89,48 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String id = userService.getCurrentUser().getUserId();
-
     // Check if user is logged in.
-    if (id == null) {
+    if (userService.getCurrentUser() == null) {
       response.sendError(
           HttpServletResponse.SC_NOT_FOUND, "You don't have permission to perform this action!");
       return;
     }
 
     // Mandatory property "name" needs to be filled out. If not, send an error.
-    if (request.getParameter("name") == null) {
+    if (request.getParameter(NAME_PROPERTY) == null) {
       response.sendError(
           HttpServletResponse.SC_NOT_FOUND, "Required field: name was not filled out.");
       return;
     }
 
+    String id = userService.getCurrentUser().getUserId();
+
     // Update properties in datastore.
     Entity profileEntity = new Entity("UserProfile", id);
-    profileEntity.setProperty("isBusiness", request.getParameter("isBusiness"));
-    profileEntity.setProperty("name", request.getParameter("name"));
-    profileEntity.setProperty("location", request.getParameter("location"));
-    profileEntity.setProperty("bio", request.getParameter("bio"));
+    profileEntity.setProperty(IS_BUSINESS_PROPERTY, getParam(IS_BUSINESS_PROPERTY, request));
+    profileEntity.setProperty(NAME_PROPERTY, getNameParam(NAME_PROPERTY, request));
+    profileEntity.setProperty(LOCATION_PROPERTY, getParam(LOCATION_PROPERTY, request));
+    profileEntity.setProperty(BIO_PROPERTY, getParam(BIO_PROPERTY, request));
 
     // Put entity in datastore.
     datastore.put(profileEntity);
 
     response.sendRedirect("/profile.html?id=" + id);
   }
+
+  public String getParam(String property, HttpServletRequest request) {
+    if (request.getParameter(property) == null) {
+      return "";
+    }
+
+    return request.getParameter(property);
+  }
+
+  public String getNameParam(String name, HttpServletRequest request) {
+    if (request.getParameter(name) == null) {
+      return "Anonymous";
+    }
+
+    return request.getParameter(name);
+  }
```

Comment:
```suggestion
  public String getNameParam(HttpServletRequest request) {
    if (request.getParameter(NAME_PROPERTY) == null) {
      return "Anonymous";
    }

    return request.getParameter(name);
  }
```
I think this would be better since this function is specifically designed to get the name property.

Reply:
Good catch! Thanks, done.